### PR TITLE
docs: Reference doc updates for permissive mTLS settings

### PR DESCRIPTION
**Pull request by pglass**
### Description

This updates the documentation with the new settings for permissive mTLS:

* New fields in config entries
* Filtering support for config entries

### Links

* https://github.com/hashicorp/consul/pull/17035
* https://github.com/hashicorp/consul/pull/17183

### PR Checklist

* [ ] updated test coverage
* [x] external facing docs updated
* [ ] appropriate backport labels added
* [x] not a security concern

### CODE REVIEW

File: website/content/docs/connect/config-entries/service-defaults.mdx [hunk 1]
```diff
@@ -10,7 +10,7 @@ This topic describes how to configure service defaults configuration entries. Th
 
 ## Configuration model
 
-The following outline shows how to format the service splitter configuration entry. Click on a property name to view details about the configuration.
+The following outline shows how to format the service defaults configuration entry. Click on a property name to view details about the configuration.
```
Comment (cthain) [inline website/content/docs/connect/config-entries/service-defaults.mdx:13]: Nice catch!

**Review by cthain — APPROVED** [review]
LGTM :+1:

**Review by trujillo-adam — APPROVED** [review]
Looks great! Just a few minor nits.

File: website/content/docs/services/discovery/dns-configuration.mdx [hunk 2]
```diff
@@ -12,37 +12,37 @@ This topic describes the default behavior of the Consul DNS functionality and ho
...
+- [`domain`](/consul/docs/agent/config/config-files#domain)
 - [`alt_domain`](/consul/docs/agent/config/config-files#alt_domain)
-- [`dns_config`](/consul/docs/agent/config/config-files#dns_config) 
+- [`dns_config`](/consul/docs/agent/config/config-files#dns_config)
 
 ### Configure WAN address translation
 By default, Consul DNS queries return a node's local address, even when being queried from a remote datacenter. You can configure the DNS to reach a node from outside its datacenter by specifying the address in the following configuration fields in the Consul agent:
 
-- [advertise-wan](/consul/docs/agent/config/cli-flags#_advertise-wan) 
-- [translate_wan_addrs](/consul//docs/agent/config/config-files#translate_wan_addrs)
```
Comment (pglass) [inline website/content/docs/services/discovery/dns-configuration.mdx:29]: I noticed Vercel was complaining about the `//` here, so I fixed this (the rest of the changes in this file are trimming whitespace, or actually maybe newline conversions).

File: website/content/docs/release-notes/consul/v1_13_x.mdx [hunk 3]
```diff
@@ -15,7 +15,7 @@ description: >-
 
 - **Enables TLS on the Envoy Prometheus endpoint**: The Envoy prometheus endpoint can be enabled when `envoy_prometheus_bind_addr` is set and then secured over TLS using new CLI flags for the `consul connect envoy` command. These commands are: `-prometheus-ca-file`, `-prometheus-ca-path`, `-prometheus-cert-file` and `-prometheus-key-file`. The CA, cert, and key can be provided to Envoy by a Kubernetes mounted volume so that Envoy can watch the files and dynamically reload the certs when the volume is updated.
 
-- **UDP Health Checks**: Adds the ability to register service discovery health checks that periodically send UDP datagrams to the specified IP/hostname and port. Refer to [UDP checks](/consul/docs//services/usage/checks#udp-checks).
+- **UDP Health Checks**: Adds the ability to register service discovery health checks that periodically send UDP datagrams to the specified IP/hostname and port. Refer to [UDP checks](/consul/docs/services/usage/checks#udp-checks).
```
Comment (pglass) [inline website/content/docs/release-notes/consul/v1_13_x.mdx:18]: Same as my other comment: this is intentional to fix the `//`

**Pull request merged**